### PR TITLE
volatile: add command method to be used in Packet.command()

### LINKED ISSUE #2828: Syntax errors in Packet.command output when using VolatileValue
Hi all,

VolatileValue class is not handled properly in Packet.command method. This results in syntax errors when using output of command method.

#### Environment

-   Scapy version: Latest (9486bb2cef76)
-   Python version: 3.7.3
-   Operating System: Debian 10

#### How to reproduce

```
>>> eval(ICMPv6NIQueryNOOP().command())                                                                                                                       
Traceback (most recent call last):

  File "/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py", line 3326, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "<ipython-input-1-b772ab6aa3ea>", line 1, in <module>
    eval(ICMPv6NIQueryNOOP().command())

  File "<string>", line 1
    ICMPv6NIQueryNOOP(nonce=<RandBin>)
                            ^
SyntaxError: invalid syntax
```

#### Expected result

This should return an instance of ICMPv6NIQueryNOOP. From v2.4.0:
```
>>> eval(ICMPv6NIQueryNOOP().command())
<ICMPv6NIQueryNOOP  |>
```

#### Related resources

A git bisect gave me this commit:
de079743169414141b24026c656eace4d33e7d78

I can try to fix this, but I am not sure what should be done. Here are solutions I could think of:

 * Generate the value (call to _fix method right?): ICMPv6NIQueryNOOP(nonce='\x21...')
 * return the string required to instance another VolatileValue: ICMPv6NIQueryNOOP(nonce=RandBin())

The first solution loses the 'randomness' for the nonce, which can be bad (?). I don't know nonce for ICMPv6 but I know they are important for security reasons in IPsec so...
The second solution can crash if the VolatileValue is not imported prior to evaluation of the command output. Which is not an issue if using run_scapy or scapy.all.

Keep me posted if you want me to fix this ;-)

### PR DESCRIPTION
**Pull request by ThomasFaivre**
When a VolatileValue subclass is used inside a Packet, the command()
method returns a string that get a Syntax Error when evaluated, e.g.:

> ICMPv6NIQueryNOOP(nonce=<RandBin>)

Add a command() method to the VolatileValue class so that it can be
handled properly. The _command_args() method is used to defined the
string used as argument for a given VolatileValue subclass.

Default values are not displayed.

Fixes #2828.

Signed-off-by: Thomas Faivre <thomas.faivre@6wind.com>

### CODE REVIEW

**Comment by ThomasFaivre** [comment]
Not sure what I can do about Codacy. The purpose of this method is to get overridden by sub-classes.
Any advice?

Also, coverage went down a little, that's because I only added command test for classes that were already being tested.

File: scapy/volatile.py [hunk 1]
```diff
@@ -337,6 +367,13 @@ def __init__(self, enum, seed=None):
         self.enum = list(enum)
         RandEnum.__init__(self, 0, len(self.enum) - 1, seed)
 
+    def _command_args(self):
+        # XXX: only outputs the list of keys, but values are irrelevant anyway
```
Comment (guedou) [inline scapy/volatile.py:371]: Could you replace `XXX` with `Note` ?

File: scapy/volatile.py [hunk 2]
```diff
@@ -394,13 +462,19 @@ def __init__(self, iptemplate="0.0.0.0/0"):
         RandString.__init__(self)
         self.ip = Net(iptemplate)
 
+    def _command_args(self):
+        if self.ip.repr == "0.0.0.0/0":
```
Comment (guedou) [inline scapy/volatile.py:466]: Could use a `_DEFAULT_IPTEMPLATE `variable as you did with `_DEFAULT_CHARS`?

**Review by guedou — CHANGES_REQUESTED** [review]
Thanks for the long PR!

**Pull request merged**